File: src/main/java/cn/jgayb/fenixplugin/FenixsFileTemplateDescriptorFactory.java
package cn.jgayb.fenixplugin;

import com.intellij.ide.fileTemplates.FileTemplateDescriptor;
import com.intellij.ide.fileTemplates.FileTemplateGroupDescriptor;
import com.intellij.ide.fileTemplates.FileTemplateGroupDescriptorFactory;
import com.intellij.openapi.util.IconLoader;

/**
 * @author yanglin
 */
@SuppressWarnings("TestInspectionTool")
public class FenixsFileTemplateDescriptorFactory implements FileTemplateGroupDescriptorFactory {

    public static final String FENIX_MAPPER_XML_TEMPLATE = "Fenixs Mapper.xml";

    @Override
    public FileTemplateGroupDescriptor getFileTemplatesDescriptor() {
        FileTemplateGroupDescriptor group = new FileTemplateGroupDescriptor("Fenixs", IconLoader.getIcon("/images/pluginIcon.svg", FenixsFileTemplateDescriptorFactory.class));
        group.addTemplate(new FileTemplateDescriptor(FENIX_MAPPER_XML_TEMPLATE, IconLoader.getIcon("/images/pluginIcon.svg", FenixsFileTemplateDescriptorFactory.class)));
        return group;
    }

}

File: src/main/java/cn/jgayb/fenixplugin/FenixsFileTemplateDescriptorFactory.kt
package cn.jgayb.fenixplugin

import com.intellij.ide.fileTemplates.FileTemplateDescriptor
import com.intellij.ide.fileTemplates.FileTemplateGroupDescriptor
import com.intellij.ide.fileTemplates.FileTemplateGroupDescriptorFactory
import com.intellij.openapi.util.IconLoader.getIcon

class FenixsFileTemplateDescriptorFactory : FileTemplateGroupDescriptorFactory {
    override fun getFileTemplatesDescriptor(): FileTemplateGroupDescriptor {
        val group = FileTemplateGroupDescriptor(
            "Fenixs", getIcon(
                "/images/pluginIcon.svg",
                FenixsFileTemplateDescriptorFactory::class.java
            )
        )
        group.addTemplate(
            FileTemplateDescriptor(
                FENIX_MAPPER_XML_TEMPLATE, getIcon(
                    "/images/pluginIcon.svg",
                    FenixsFileTemplateDescriptorFactory::class.java
                )
            )
        )
        return group
    }

    companion object {
        const val FENIX_MAPPER_XML_TEMPLATE: String = "Fenixs Mapper.xml"
    }
}